Unable to install Rstan with R 3.4.4

Using install.packages function, I get compilation errors. I am using R 3.4.4 on Ubuntu 16.04.5 LTS.

A portion of the output showing the errors:

> In file included from chains.cpp:19:0:
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp: In function 'typename boost::math::tools::promote_args<T>::type stan::math::mean(const std::vector<T>&)':
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:63: error: an array reference cannot appear in a constant-expression
>    Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>> m(&v[0], v.size());
>                                                                ^
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:63: error: '&' cannot appear in a constant-expression
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:68: error: '.' cannot appear in a constant-expression
>    Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>> m(&v[0], v.size());
>                                                                     ^
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:73: error: a function call cannot appear in a constant-expression
>    Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>> m(&v[0], v.size());
>                                                                          ^
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:74: error: a function call cannot appear in a constant-expression
>    Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>> m(&v[0], v.size());
>                                                                           ^
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:74: error: template argument 3 is invalid
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:74: error: template argument 4 is invalid
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:74: error: template argument 6 is invalid
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:24:27: error: template argument 1 is invalid
>    Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>> m(&v[0], v.size());
>                            ^
> /usr/local/lib/R/site-library/StanHeaders/include/stan/math/prim/mat/fun/mean.hpp:25:10: error: 'm' was not declared in this scope
>    return m.mean();
>           ^

--- a/rstan/rstan/R/misc.R
+++ b/rstan/rstan/R/misc.R
@@ -364,7 +364,7 @@ config_argss <- function(chains, iter, warmup, thin,
   if (thin < 1 || thin > iter)
     stop("parameter 'thin' should be a positive integer less than 'iter'")
   warmup <- max(0, as.integer(warmup))
-  if (warmup > iter)
+  if (warmup >= iter)
     stop("parameter 'warmup' should be an integer less than 'iter'")
   chains <- as.integer(chains)
   if (chains < 1)

--- a/rstan/rstan/man/stan.Rd
+++ b/rstan/rstan/man/stan.Rd
@@ -84,7 +84,7 @@ stan(file, model_name = "anon_model", model_code = "", fit = NA,
     iterations per chain. If step-size adaptation is on (which it is by default), 
     this also controls the number of iterations for which adaptation is run (and
     hence these warmup samples should not be used for inference). The number of 
-    warmup iterations should not be larger than \code{iter} and the default is 
+    warmup iterations should be smaller than \code{iter} and the default is
     \code{iter/2}.}
     
   \item{chains}{A positive integer specifying the number of Markov chains. 

--- a/rstan/rstan/man/stanmodel-method-sampling.Rd
+++ b/rstan/rstan/man/stanmodel-method-sampling.Rd
@@ -59,7 +59,7 @@
     iterations per chain. If step-size adaptation is on (which it is by default), 
     this also controls the number of iterations for which adaptation is run (and
     hence these warmup samples should not be used for inference). The number of 
-    warmup iterations should not be larger than \code{iter} and the default is 
+    warmup iterations should be smaller than \code{iter} and the default is
     \code{iter/2}.}
 
   \item{thin}{A positive integer specifying the period for saving samples. 

